Test files are persisting between branches

## Description

Files created in the jenkins workspace during a test run triggered by a push to the `run-integration-tests` branch are persisting between runs, causing test failures. 

The following failure, from https://ci.vpn1.moz.works/blue/organizations/jenkins/bedrock_multibranch_pipeline/detail/run-integration-tests/380/pipeline#step-163-log-531 is `FAILED tests/functional/firefox/privacy/test_products.py::test_download_button_displayed` while running commit b707c247e but as can be seen in https://github.com/mozilla/bedrock/tree/b707c247e/tests/functional/firefox this directory and file do not exist in that commit. 

Looking at the working directory, `/var/lib/jenkins/jobs/bedrock_multibranch_pipeline/branches/run-integration-tests/workspace@5`, I found that there were a large number of files from previous commits (including the culprit above) that should have been cleaned out during the `make clean build` step of the pipeline, which runs a `git clean -f` command. Running this command manually did in fact remove those files from the workspace (and then I iterated through the other workspaces to do the same) but the docker images previously built from this working directory still contain the files. To address the specific failure above, I ran `docker rmi mozorg/bedrock_test:b707c247e32f7274d703c2b5a5c064c51f30dee2 -f`.

--- a/Makefile
+++ b/Makefile
@@ -75,8 +75,8 @@ clean:
 	-rm -rf static_build/
 #	state files
 	-rm -f .docker-build*
-# clean untracked files
-	git clean -f
+# clean untracked files & directories
+	git clean -d -f
 
 lint: .docker-build-pull
 	${DC} run test flake8

